### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by ShyLee873**

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class Comment < ApplicationRecord
+end
```
Comment (kostik700015) [inline app/models/comment.rb:2]: Why did you remove these lines? I think we need them for the polymorphic association. 
```
`belongs_to :commentable, polymorphic: true`
`validates :body, presence: true`
```
For the polymorphic association you should add them to make association works.

Also we may need to add `has_many :comments` to User Model so User can have many comments

**Review by kostik700015 — CHANGES_REQUESTED** [review]
I think that we actually don't need a polymorphic association between Post and Comment models. The polymorphic association is used when the `Model-A` can `belongs_to` either `Model-B` or `Model-C`. In our case Comments can belong both User and Post.
@jacobgwilliams , @Xxyumi-hub  what do you think?

File: app/models/comment.rb (hunk 1 above)
Comment (ShyLee873) [reply app/models/comment.rb:2]: The polymorphic association  is so that the Comment model can be integrated into both the post and events as discussed. I'm not sure what happened to the comment model, I had to pull stuff from the repo before I could make my changes. I'll fix it.

**Comment by kostik700015** [comment]
I also got an error when ran `rails db:migrate`. I think it happened because it tried to delete non-existing yet table 'comments'(in order before '**comments**' table was created). After I deleted the migration file `20220201193625_drop_comments.rb` I could run migration and '**comments**' was created

**Comment by ShyLee873** [comment]
I'll have to push the changes after this is merged. Before, my local branch was behind for some reason and I had to take some corrective measures to fix the issue. `the belongs_to :commentable, polymorphic: true` line was overlooked in the repair. I'll have to add it back with the next push to avoid repeating the same mistakes.

**Comment by ShyLee873** [comment]
> I also got an error when ran `rails db:migrate`. I think it happened because it tried to delete non-existing yet table 'comments'(in order before '**comments**' table was created). After I deleted the migration file `20220201193625_drop_comments.rb` I could run migration and '**comments**' was created

Yep. this is all corrected with this latest PR. Again, my branch was behind and things were changed and deleted on my end. With this PR, my new comments table is functional.

File: app/models/comment.rb (hunk 1 above)
Comment (kostik700015) [reply app/models/comment.rb:2]: > The polymorphic association is so that the Comment model can be integrated into both the post and events as discussed. 

- I totally agree with it.  I was confused when noticed empty comment model.  Sure, after pulling last changes you will have opportunity to do that as we discussed with polymorphic model.  Let me know if you have any questions.

**Comment by ShyLee873** [comment]
I'll just close and do it all again after I get my kids to bed. No worries.
